tests/test_helper/bats/bgp.bats

```diff
@@ -0,0 +1,116 @@
+
+    # Ensure TEST_CONTAINERS is populated, otherwise the tests below will
+    # provide false positive results.
+    assert [ -n "$TEST_CONTAINERS" ]
+    assert [ -n "$BGP_PEERS" ]
+    assert [ -n "$MICROOVN_BGP_CONTAINER" ]
+
+
+}
+
+teardown() {
+    echo "# ($MICROOVN_BGP_CONTAINER) Disabling MicroOVN BGP" >&3
+    lxc_exec "$MICROOVN_BGP_CONTAINER" "microovn disable bgp"
+
+    # Check that NB resources were cleaned up
+    run lxc_exec "$MICROOVN_BGP_CONTAINER" "microovn.ovn-nbctl show"
+    assert_success
+    assert_output ""
+
+    # Check that OVS Bridges were cleaned up
+    run lxc_exec "$MICROOVN_BGP_CONTAINER" "microovn.ovs-vsctl find bridge name!=br-int"
+    assert_success
+    assert_output ""
+
+    # Check that OVS Ports were cleaned up
+    run lxc_exec "$MICROOVN_BGP_CONTAINER" "microovn.ovs-vsctl find interface type!=geneve name!=br-int"
+    assert_success
+    assert_output ""
+
+    # Check that OVN Bridge mappings were cleaned up
+    run lxc_exec "$MICROOVN_BGP_CONTAINER" "microovn.ovs-vsctl get Open_vSwitch . external-ids:ovn-bridge-mappings"
+    assert_failure
+    assert_output 'ovs-vsctl: no key "ovn-bridge-mappings" in Open_vSwitch record "." column external_ids'
+
+    # Restart FRR in peer containers to clean up running config
+    local container
+    for container in $BGP_PEERS; do
+        echo "# ($container) Resetting FRR running configuration" >&3
+        lxc_exec "$container" "systemctl restart frr"
+    done
+}
+
+tls_cluster_register_test_functions() {
+    bats_test_function \
+        --description "OVN with multiple BGP peers" \
+        -- bgp_multiple_peers
+
+    bats_test_function \
+        --description "OVN with single BGP peer" \
+        -- bgp_single_peer
+}
+
+bgp_multiple_peers() {
+    local container
+    for container in $BGP_PEERS; do
+        echo "# Starting BGP in $container on interface $BGP_CONTAINER_IFACE" >&3
+        frr_start_bgp_unnumbered "$container" "$BGP_CONTAINER_IFACE" 1
+    done
+
+    local BGP_NET_1_IP="10.0.10.1/24"
+    local BGP_NET_2_IP="10.0.20.1/24"
+    local vrf="10"
+    local bgp_iface_1="$OVN_CONTAINER_NET_1_IFACE-bgp"
+    local bgp_iface_2="$OVN_CONTAINER_NET_2_IFACE-bgp"
+    local vrf_device="ovnvrf$vrf"
+
+    echo "# Enabling MicroOVN BGP in $MICROOVN_BGP_CONTAINER"
+    local external_connections="$OVN_CONTAINER_NET_1_IFACE:$BGP_NET_1_IP,$OVN_CONTAINER_NET_2_IFACE:$BGP_NET_2_IP"
+    lxc_exec "$MICROOVN_BGP_CONTAINER" "microovn enable bgp \
+        --config ext_connection=$external_connections \
+        --config vrf=$vrf"
+
+    microovn_start_bgp_unnumbered "$MICROOVN_BGP_CONTAINER" "$bgp_iface_1" 1 "$vrf_device"
+    microovn_start_bgp_unnumbered "$MICROOVN_BGP_CONTAINER" "$bgp_iface_2" 1 "$vrf_device"
+
+
+    for container in $BGP_PEERS; do
+        echo "# ($MICROOVN_BGP_CONTAINER) waiting on established BGP with $container" >&3
+        wait_until "microovn_bgp_established $MICROOVN_BGP_CONTAINER $vrf_device $container"
+    done
+}
```

Comment:
Should we also check redistribution of routes? At this stage it could be as simple as adding a couple of NAT entries and checking that our bare bones BGP config allows the route to this address show up in the routing table in the remote BGP peer.

Reply:
This was not yet added.